### PR TITLE
TRUNK-5564:PersonAddressValidator - Throws exception instead of logging

### PR DESCRIPTION
**Pull request by jwnasambu**
## Issue I worked on
https://issues.openmrs.org/browse/TRUNK-5564
I coded the PersonAddressValidator in a way that avoids throwing exceptions

### CODE REVIEW

File: api/src/main/java/org/openmrs/validator/PersonAddressValidator.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public void validate(Object object, Errors errors) {
 		}
 		
 		if (object == null) {
-			throw new IllegalArgumentException("The personAddress object should not be null");
+			errors.reject("error.Address");
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/validator/PersonAddressValidator.java]: according to the needs of your ticket,i see you need to use some logging and cannot see it here

**Comment by jwnasambu** [comment]
@ayesh I have made the changes. Thanks

File: api/src/main/java/org/openmrs/validator/PersonAddressValidator.java [hunk 2]
```diff
@@ -64,7 +64,8 @@ public void validate(Object object, Errors errors) {
 		}
 		
 		if (object == null) {
-			throw new IllegalArgumentException("The personAddress object should not be null");
+			log.error("Address object contains null value"); 
+			return;
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/validator/PersonAddressValidator.java]: do we really need the return i see your method is a void!!

**Comment by Akayeshmantha** [comment]
Nice work @jwamalwa 

@dkayiwa  can we merge this PR

**Comment by dkayiwa** [comment]
Can you follow the same pattern used by other validators? https://github.com/openmrs/openmrs-core/tree/master/api/src/main/java/org/openmrs/validator

**Comment by Akayeshmantha** [comment]
@dkayiwa 

The ticket says not to throw the exception.

For example this validator does the same as @jwamalwa  does for this validator and it's good to have a logging message for reference as well so I don't see anything to change here : 

https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/validator/PatientValidator.java

**Comment by dkayiwa** [comment]
@Akayeshmantha did you look at the validators in the link that i gave?

**Comment by Akayeshmantha** [comment]
Hmm yes, @dkayiwa . The one I posted in the previous comment is also a validator in the same link

**Comment by dkayiwa** [comment]
@Akayeshmantha how many validators did you look at?

**Comment by jwnasambu** [comment]
@dkayiwa, @Akayeshmantha I have made some changes kindly review my PR and feel free to advice me.

**Comment by dkayiwa** [comment]
@jwnasambu did you take a look at the validators that i pointed to?

**Comment by jwnasambu** [comment]
@dkayiwa yes I did only that they are several and am still going through to understand.

**Comment by Akayeshmantha** [comment]
Hi @dkayiwa 

Sure will go through

**Pull request closed without merging**